core/src/main/resources/error/error-classes.json

```diff
@@ -933,6 +933,12 @@
     ],
     "sqlState" : "42604"
   },
+  "INVALID_TYPE_FOR_FILTER_EXPR" : {
+    "message" : [
+      "Filter expression '<filter>' of type <type> is not a boolean."
+    ],
+    "sqlState" : "428H2"
```

Comment:
FYI: referred to https://github.com/apache/spark/tree/master/core/src/main/resources/error